src/pages/blog.astro

```diff
@@ -10,6 +10,6 @@ const allPosts = await getCollection("blog");
 <BaseLayout pageTitle={pageTitle}>
   <p>This is where I will post about my journey learning Astro.</p>
   <ul>
-    {allPosts.map((post:any) => <BlogPost url={"/posts/" + post.slug} title={post.data.title} />)}
+    {allPosts.map((post:any) => <BlogPost url={"/posts/" + post.id} title={post.data.title} />)}

```

Comment:
To be [consistent with the docs](https://docs.astro.build/en/tutorial/6-islands/4/#replace-importmetaglob-with-getcollection) and since it's not necessary with content collections:
```suggestion
    {allPosts.map((post) => <BlogPost url={"/posts/" + post.id} title={post.data.title} />)}

```